src/yz_kv.erl

```diff
@@ -460,3 +463,139 @@ is_owner_or_future_owner(P, Node, Ring) ->
 -spec is_service_up(atom(), node()) -> boolean().
 is_service_up(Service, Node) ->
     lists:member(Service, riak_core_node_watcher:services(Node)).
+
+%% @private
+%%
+%% @doc Check if object has 2.0 CRDT datatype entry or property for
+%%      strong consistency.
+-spec is_datatype_or_consistent(obj()) -> boolean().
+is_datatype_or_consistent(Obj) ->
+    case riak_kv_crdt:is_crdt(Obj) of
+        false ->
+            is_strongly_consistent(Obj);
+        true ->
+            true;
+        _ ->
+            false
+    end.
+
+%% @private
+%%
+%% @doc Check if object has property for strong consistency.
+-spec is_strongly_consistent(riak_object:riak_object()) -> boolean().
+is_strongly_consistent(Obj) ->
+    Bucket = riak_object:bucket(Obj),
+    case riak_kv_util:consistent_object(Bucket) of
+        true ->
+            true;
+        _ ->
+            false
+    end.
+
+
+%% @private
+%%
+%% @doc Set yz_solr:index delete operation(s) on write_reason.
+-spec delete_operation(obj(), put|handoff|anti_entropy, [doc()], bkey(), lp()) ->
+                              []|[{id, _}]|[{siblings, _}].
+delete_operation(Obj, put, Docs, BKey, LP) ->
```

Comment:
Understood. I think we should fix the underlying problem in that case, but that should be ok for now.